Code/Framework/AzCore/Tests/AZStd/ChronoTests.cpp

```diff
@@ -181,7 +186,507 @@ namespace UnitTest
+
+            // Day 32 - not ok
+            testDay = AZStd::chrono::day{ 32 };
+            EXPECT_FALSE(testDay.ok());
+        }
+
+        TEST_F(DurationTest, Day_Comparison_Succeeds)
+        {
+            constexpr AZStd::chrono::day testDay1{ 13 };
+            constexpr AZStd::chrono::day testDay2{ 27 };
+            static_assert(testDay1 == testDay1);
+            static_assert(testDay1 != testDay2);
+            static_assert(testDay1 < testDay2);
+            static_assert(testDay2 > testDay1);
+            static_assert(testDay1 <= testDay2);
+            static_assert(testDay2 >= testDay1);
+        }
+
+        TEST_F(DurationTest, Day_Arithmetic_Succeeds)
+        {
+            constexpr AZStd::chrono::day testDay1{ 20 };
+            constexpr AZStd::chrono::day testDay2{ 25 };
+            constexpr AZStd::chrono::days days{ 15 };
+            static_assert(testDay1 + days == AZStd::chrono::day{ 35 });
+            static_assert(testDay2 - days == AZStd::chrono::day{ 10 });
+            static_assert(testDay2 - testDay1 == AZStd::chrono::days{ 5 });
+            static_assert(testDay1 - testDay2 == AZStd::chrono::days{ -5 });
+        }
+
+        // BIG NOTE: chrono::months with an s is a chrono::duration counting months
+        // chrono::month without an s is a abstraction of a month
+        TEST_F(DurationTest, Month_CanRepresent_MonthOfYear)
+        {
+            AZStd::chrono::month testMonth{};
+            // A month is only valid between 1 and 12 inclusive
+            // default constructs to 0
+            EXPECT_FALSE(testMonth.ok());
+
+            // pre-increment / pre-decrement
+            auto updatedMonth = ++testMonth;
+            EXPECT_EQ(testMonth, updatedMonth);
+
+            updatedMonth = --testMonth;
+            EXPECT_EQ(testMonth, updatedMonth);
+
+            // post-increment / post-decrement
+            auto oldMonth = testMonth++;
+            EXPECT_NE(oldMonth, testMonth);
+
+            oldMonth = testMonth--;
+            EXPECT_NE(oldMonth, testMonth);
+
+            // Month 1 - ok
+            testMonth = AZStd::chrono::month{ 1 };
+            EXPECT_TRUE(testMonth.ok());
+
+            // Month 12 - ok
+            testMonth = AZStd::chrono::month{ 12 };
+            EXPECT_TRUE(testMonth.ok());
+
+            // Month 13 - not ok
+            testMonth = AZStd::chrono::month{ 13 };
+            EXPECT_FALSE(testMonth.ok());
+        }
+
+        TEST_F(DurationTest, Month_Comparison_Succeeds)
+        {
+            constexpr AZStd::chrono::day testMonth1{ 13 };
+            constexpr AZStd::chrono::day testMonth2{ 27 };
+            static_assert(testMonth1 == testMonth1);
+            static_assert(testMonth1 != testMonth2);
+            static_assert(testMonth1 < testMonth2);
+            static_assert(testMonth2 > testMonth1);
+            static_assert(testMonth1 <= testMonth2);
+            static_assert(testMonth2 >= testMonth1);
+        }
```

Comment:
General feedback...

I don't know if we have guidance on this, but writing tests that only contain compile-time checks/asserts seems like a different class of test.  Like, are these `TEST_F` tests being "run"?  They aren't doing anything at runtime, don't have any `EXPECT`/`ASSERT` as is usually seen.  I'm wondering if there's a better place for static tests, or something in the gtest framework that could be used to mark them as compile-time-only tests.

Reply:
Any marking of these test as compile-time-only is only for documentation as it wouldn't effect correctness.

This static asserts can technically be in a regular function that is invoked at compile time(Though it may get dead0stripped if it doesn't have any side effects.).

Reply:
I would think that as long as all the static_asserts are guaranteed being evaluated, it shouldn't matter if the function they live in is dead-stripped.  In fact, we might prefer it to be stripped from the binaries.  If it compiles, then the static_asserts have done their job.  I just wouldn't want to have to pay for setup/teardown of the fixture for what's essentially an empty function.